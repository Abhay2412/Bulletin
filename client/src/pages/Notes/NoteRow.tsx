import {Box, Button, Typography, TextField} from "@mui/material"
import {useState} from 'react'
import React from 'react'
import { StringLiteralLike } from "typescript";

interface Props {
    id: string;
    name: string;
    date: string;
    content: string;
    onDelete: (arg0: string)=>void;
    onEdit: (arg0: string, arg1: string, arg2: string, arg3: string)=>void;
}
interface NewState {
  name: string;
  content: string;
  date: string;
}

export default function NoteRow({name, date, content, id, onDelete, onEdit}: Props) {
    const [popupEditState, setPopupEditState] = useState(false);
    const [popupOpenState, setPopupOpenState] = useState(false);
    const [newValues, setNewValues] = React.useState<NewState>({
      name: name,
      content: content,
      date: date,
    });
  const handleChange =
  (prop: keyof NewState) => (event: React.ChangeEvent<HTMLTextAreaElement>) => {
    setNewValues({ ...newValues, [prop]: event.target.value });
  };
  const handleSave = () =>{
    setPopupEditState(false);
    onEdit(id, newValues.name, newValues.content, newValues.date);
  }
  return (
    <>
    {popupEditState ? (
         <div className="overlay" style={{paddingTop: "15%"}}>
         <div className="overlayBox">
           <div className="noteHeader">
             <Typography variant="h6">
               Note Name
             </Typography>
             <TextField  variant="standard" defaultValue={name} onChange={handleChange("name")}></TextField>
             <Button onClick={()=>setPopupEditState(false)}>
               <Typography variant="h5">
                   X
               </Typography>
             </Button>
           </div>
           <div className="noteContent">
               <textarea defaultValue={content} onChange={handleChange("content")}></textarea>
           </div>
           <div className="saveDiv">
                    <Button className="saveButton" onClick={handleSave}>
                        Save Note
                    </Button>
           </div>
         </div>
     </div>
    ) : null}

    {popupOpenState ? (
            <div className="overlay" style={{paddingTop: "20%"}}>
                <div className="overlayBox">
                  <div className="noteHeader">
                    <Typography variant="h4">
                      {name}
                    </Typography>
                    <Button onClick={()=>setPopupOpenState(false)}>
                    <Typography variant="h5">
                        X
                    </Typography>
                    </Button>
                  </div>
                  <div className="noteContent">
                    {content}
                  </div>
                </div>
            </div>
        ) : null}

      <Box className="notesTableRow tableCellsFormatting noteNameCell" width="70%">
        <Box>
          {name}
        </Box>
        <Box>
          <Button variant="outlined" onClick={()=>setPopupOpenState(true)}>
            Open
          </Button>
        </Box>
      </Box>
      <Box className="notesTableRow tableCellsFormatting" width="20%">
        {date}
      </Box>
      
      <Box className="notesTableRow" width="5%">
        <Button onClick={()=>onDelete(id)}>X</Button>
      </Box>
      <Box className="notesTableRow" width="5%">
        <Button sx={{ textDecoration: "underline" }} onClick={()=>setPopupEditState(true)}>edit</Button>
      </Box> 
    </>
  )
}

