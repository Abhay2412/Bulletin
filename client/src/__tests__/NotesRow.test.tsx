import React, { useState } from "react";
import NoteRow from "../pages/Notes/NoteRow";
import { configure, mount, ReactWrapper, shallow } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import renderer from "react-test-renderer";
import { store } from "../store";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

configure({ adapter: new Adapter() });

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useParams: () => ({
    board_id: "g7HGuvfnGe3nhQgAF8qY",
  }),
}));

describe("Testing <NotesRow /> Component", () => {
  it("should render NotesRow properly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <NoteRow
              id="32iwjsdikd09a"
              name="Test Note"
              date="2022-03-28"
              content="test not content"
              onDelete={() => {}}
              onEdit={() => {}}
            />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render NotesRow properly with open modal", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <NoteRow
              id="32iwjsdikd09a"
              name="Test Note"
              date="2022-03-28"
              content="test not content"
              onDelete={() => {}}
              onEdit={() => {}}
              popupOpen={true}
            />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should open a view modal", () => {
    const wrapper: ReactWrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <NoteRow
            id="32iwjsdikd09a"
            name="Test Note"
            date="2022-03-28"
            content="test not content"
            onDelete={() => {}}
            onEdit={() => {}}
            popupOpen={false}
          />
        </BrowserRouter>
      </Provider>
    );

    wrapper.find("button").find("#view-note-dialog-button").simulate("click");
    expect(wrapper.find(".overlayBox").exists()).toBeTruthy();
  });

  it("should open an editable modal", () => {
    const wrapper: ReactWrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <NoteRow
            id="32iwjsdikd09a"
            name="Test Note"
            date="2022-03-28"
            content="test not content"
            onDelete={() => {}}
            onEdit={() => {}}
            popupOpen={false}
          />
        </BrowserRouter>
      </Provider>
    );

    wrapper.find("button").find("#edit-note-dialog-button").simulate("click");
    expect(wrapper.find(".overlayBox").exists()).toBeTruthy();
  });

  describe("should have edit modal open", () => {
    const wrapper: ReactWrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <NoteRow
            id="32iwjsdikd09a"
            name="Test Note"
            date="2022-03-28"
            content="test not content"
            onDelete={() => {}}
            onEdit={() => {}}
            popupOpen={false}
          />
        </BrowserRouter>
      </Provider>
    );

    beforeAll(() => {
      wrapper.find("button").find("#edit-note-dialog-button").simulate("click");
    });

    it("should change note title value", () => {
      wrapper
        .find("input")
        .find("#edit-note-modal-name-field")
        .simulate("change", {
          target: { value: "Test Note" },
        });
      expect(
        wrapper.find("input").find("#edit-note-modal-name-field").get(0).props
          .value
      ).toEqual("Test Note");
    });

    it("should change note description value", () => {
      wrapper
        .find("textarea")
        .find("#edit-note-modal-content-field")
        .simulate("change", {
          target: { value: "Test Description" },
        });
      expect(
        wrapper.find("textarea").find("#edit-note-modal-content-field").get(0)
          .props.value
      ).toEqual("Test Description");
    });

    it("should close modal", () => {
        wrapper.find("button").find("#close-note-edit-modal-button").simulate("click");
        expect(wrapper.find(".overlayBox").exists()).toBeFalsy();
      });
  });
});
