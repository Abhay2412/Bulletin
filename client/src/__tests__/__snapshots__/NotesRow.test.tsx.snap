// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <NotesRow /> Component should render NotesRow properly 1`] = `
Array [
  <div
    className="notesTableRow tableCellsFormatting noteNameCell MuiBox-root css-6gs9hh"
  >
    <div
      className="MuiBox-root css-0"
    >
      Test Note
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <button
        className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        id="view-note-dialog-button"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Open
      </button>
    </div>
  </div>,
  <div
    className="notesTableRow tableCellsFormatting MuiBox-root css-ufscxh"
  >
    2022-03-28
  </div>,
  <div
    className="notesTableRow MuiBox-root css-ndbrsi"
  >
    <button
      className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      X
    </button>
  </div>,
  <div
    className="notesTableRow MuiBox-root css-ndbrsi"
  >
    <button
      className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-taz78y-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      id="edit-note-dialog-button"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      edit
    </button>
  </div>,
]
`;

exports[`Testing <NotesRow /> Component should render NotesRow properly with open modal 1`] = `
Array [
  <div
    className="overlay"
    style={
      Object {
        "paddingTop": "20%",
      }
    }
  >
    <div
      className="overlayBox"
    >
      <div
        className="noteHeader"
      >
        <h4
          className="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
        >
          Test Note
        </h4>
        <button
          className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <h5
            className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          >
            X
          </h5>
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        className="noteContent"
      >
        test not content
      </div>
    </div>
  </div>,
  <div
    className="notesTableRow tableCellsFormatting noteNameCell MuiBox-root css-6gs9hh"
  >
    <div
      className="MuiBox-root css-0"
    >
      Test Note
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <button
        className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        id="view-note-dialog-button"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Open
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  <div
    className="notesTableRow tableCellsFormatting MuiBox-root css-ufscxh"
  >
    2022-03-28
  </div>,
  <div
    className="notesTableRow MuiBox-root css-ndbrsi"
  >
    <button
      className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      X
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>,
  <div
    className="notesTableRow MuiBox-root css-ndbrsi"
  >
    <button
      className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-taz78y-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      id="edit-note-dialog-button"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      edit
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>,
]
`;
